src/esn.inbox.libs/app/services/mailboxes/mailboxes-service.js

```diff
@@ -73,6 +74,33 @@ angular.module('esn.inbox.libs')
       return mailbox;
     }
 
+    function mailboxtoTree(arr) {
```

Comment:
Sorry if this is a none sens in js but I would consider this to be quite a bunch of mutable stuff.

We (java / scala backend devs) would tend to try to write this in an immutable way with some kind of collect.

Reply:
for me that's the good way in JS ;)

Reply:
Don't feed the troll ;-)